odabot/cli.py

```diff
@@ -4,32 +4,39 @@
 import logging
 import os
 import re
+import shutil
 import time
 import yaml
 import tempfile
-import click
-import subprocess
+import subprocess as sp
 import requests
 from datetime import datetime
 import sys 
 import traceback
+import xml.etree.ElementTree as ET
 
-import markdown
-import rdflib
-
-from nb2workflow.deploy import build_container, deploy_k8s, ContainerBuildException
-from nb2workflow import version as nb2wver
-#from nb2workflow.validate import validate, patch_add_tests, patch_normalized_uris
-from mmoda_tab_generator.tab_generator import MMODATabGenerator
-
-from .markdown_helper import convert_help
+import click
+from dynaconf import Dynaconf
 
 logger = logging.getLogger()
 
-from dynaconf import Dynaconf
+try:
+    import markdown
+    import rdflib
+    from nb2workflow.deploy import build_container, deploy_k8s, ContainerBuildException
+    from nb2workflow import version as nb2wver
+    #from nb2workflow.validate import validate, patch_add_tests, patch_normalized_uris
+    from mmoda_tab_generator.tab_generator import MMODATabGenerator
+    from .markdown_helper import convert_help
+except ImportError:
+    logger.warning('Deployment dependencies not loaded')
+    
+try:
+    from nb2workflow.galaxy import to_galaxy    
+    import frontmatter
+except ImportError:
+    logger.warning('Galaxy dependencies not loaded')
 
-# `envvar_prefix` = export envvars with `export DYNACONF_FOO=bar`.
-# `settings_files` = Load this files in the order.
 
 renkuapi = "https://gitlab.renkulab.io/api/v4/"
```

Comment:
not added here, but shouldn't this be in config? Especially if we want to use different gitlabs.

Reply:
will make a separate small pr